torch/csrc/dynamo/eval_frame.c

```diff
@@ -745,17 +893,20 @@ static PyObject* _custom_eval_frame(
     return NULL;
   } else if (result != Py_None) {
     DEBUG_TRACE("create cache %s", get_frame_name(frame));
-    extra = create_cache_entry(extra, result);
+    CacheEntry* new_cache_entry = create_cache_entry(cache_entry, result);
```

Comment:
This is also technically a violation of borrow semantics, as `cache_entry` came from a borrow. Suggest `extra->cache_entry` instead.